website/docs/tasks/compile.md

```diff
@@ -0,0 +1,76 @@
+
+The Michelson contracts produced by the compilers will be created in the `/artifacts` directory
+
+:::note
+Contracts must be added to the [Contract Registry](/docs/tasks/add-contract) (`taq add-contract example.mligo`) in order to be compiled
+:::
+
+## Plugin Implementations
+
+This task is implemented by the following plugins:
+
+| Plugin Name                  | Description                       |
+| ---------------------------- | --------------------------------- |
+| `@taqueria/plugin-ligo`      | Compiles Ligo contracts           |
+| `@taqueria/plugin-smartpy`   | Compiles SmartPy contracts        |
+| `@taqueria/plugin-archetype` | Compiles Archetype contracts      |
+
+### Command
+
+```shell
+taq compile <path> 
+```
+
+### Task Details
+
+| Task Name      | Command                       | Type                | Description                                           | 
+| -------------- | ----------------------------- | ------------------- | ----------------------------------------------------- |
+| `compile`      | `taq compile [path]`          | Plugin - compilers  | Compiles one, multiple, or all contracts in a project |
+
+### Command-Line Arguments
+
+| Argument     | Required | Description                                            | Example Usage                                         |
+| ------------ | -------- | ------------------------------------------------------ | ----------------------------------------------------- |
+| `<path>`     | No       | Path to one or more Ligo, SmartPy, or Archetype files  | `taq compile ./contracts/my-contract.jsligo`          |
+| `--plugin`   | No       | Compiles contracts using the compiler plugin specified | `taq compile --plugin archetype`                      |
+|  `-e`        | No       | The entry point that will be compiled                  | `taq compile ./contracts/test.mligo -e transferTo`    |
+|  `-s`        | No       | The syntax used in the contract                        | `taq compile -s jsligo`                               |      
+|  -i          | No       | Enable type inference                                  | `taq compile ./contracts/counter.mligo -i`            |  
+
+### Usage
+
+| Description                 | Command                            | Behaviour                                                                     |
+| --------------------------- | ---------------------------------- | ----------------------------------------------------------------------------- |
+| Compile all contracts       | `taq compile`                      | Compiles all Ligo, SmartPy, and Archetype files in the `/contracts` directory |
+| Compile one contract        | `taq compile [path]`               | Compiles the contract at the given path                                       |
+| Compile Ligo contracts      | `taq compile --plugin ligo`        | Compiles Ligo contracts using the Ligo compiler                               |
```

Comment:
We may want to note that the plugin commands are only available when multiple compilation plugins are installed